components/local-app/pkg/auth/auth.go

```diff
@@ -120,24 +120,29 @@ const html = `
 </html>`
 
 // NOTE: the port ranges all need to be valid redirect URI's in the backend
-const STARTING_PORT_NUM = 63110
-const ENDING_PORT_NUM = 63120
+const (
+	StartingPortNum = 63110
+	EndingPortNum   = 63120
+)
 
 // Login walks through the login flow for obtaining a Gitpod token
 func Login(ctx context.Context, opts LoginOpts) (token string, err error) {
 	// Try a range of ports for local redirect server
 	var rl net.Listener
 	var port int
-	for port = STARTING_PORT_NUM; port < ENDING_PORT_NUM || rl == nil; port++ {
+	for port = StartingPortNum; port < EndingPortNum || rl == nil; port++ {
 		rl, err = net.Listen("tcp4", fmt.Sprintf("127.0.0.1:%d", port))
 		if err != nil {
 			logrus.Infof("Could not open port:%d got error:%s\nTrying next port\n", port, err)
 			continue
 		}
-		defer rl.Close()
```

Comment:
`defer` inside a for loop gets only queued up and executed once the function exits. This way, we're releasing the port immediately.

Reply:
We should use `defer`, because we use `rl` below.

To make code more readable, we can

1. `break` after listening to one port success and remove condition `rl == nil`
2. defer close `rl` after check `rl` is not nil (below line 144)

I pushed a commit, feel free to revert or rebase

Reply:
Thanks, that makes sense. I'll pull the listener acquistion into a seperate method just to clean up the continue/break a bit.